File: Royaleagle73/A2_Q2.clj
(defn eligible [record] (
                        or (and 
                            (> (/ (get record 1) (get record 3)) 4)
                            (> (get record 2) 2))
                         (and
                          (< (/ (get record 1) (get record 3)) 4)
                          (> (get record 2) 4))
))

(defn man_of_the_match [records] (loop [index 0
                                        man_of_the_match_final "None"
                                        final_records {}] (
                                                              if (= index (count records))
                                                              [final_records man_of_the_match_final]
                                                                    (recur (inc index) (
                                                                                       if(= (eligible (get records index)) true)
                                                                                        (get (get records index) 0)
                                                                                        man_of_the_match_final
                                                                    )
                                                                           (assoc final_records (keyword (get (get records index) 0)) [(get (get records index) 1)
                                                                                                                                       (get (get records index) 2)
                                                                                                                                       (get (get records index) 3)]))
                                        )))

(defn main [] (do (def records [["Arun" 25 3 7]
                                ["Nihal" 50 5 12]
                                ["Sanjay" 10 4 3]])
                  (man_of_the_match records)))
(main)

File: Bhawna97/A2_Q2.py
def man_of_the_match(records):
    for i in records:
        strike_rate = i[1]/i[3]
        if strike_rate >4 and i[2]>2:
            return i[0]
        elif strike_rate >2 and strike_rate<4 and i[2]>4:
            return i[0]

records = [['Arun',25,3,7],['Nihal',50,5,12],['Sanjay',10,4,3]]
dic = {}
for i in range(len(records)):
    dic[records[i][0]]= []
    dic[records[i][0]].append(records[i][1:])
    
print(dic,"\n",man_of_the_match(records))
